core/opendaq/reader/src/tail_reader_impl.cpp

```diff
@@ -131,47 +152,74 @@ ErrCode TailReaderImpl::readData(TailReaderInfo& info)
         info.offset = cachedSamples - info.remainingToRead;
 
     ErrCode errCode = OPENDAQ_SUCCESS;
-    for (const auto& dataPacket : packets)
+    size_t readCachedSamples = 0;
+
+    for (auto it = packets.begin(); it != packets.end();)
     {
-        errCode = readPacket(info, dataPacket);
-        if (OPENDAQ_FAILED(errCode))
+        const auto & packet = *it;
+        if (packet.getType() == PacketType::Event)
         {
+            handleDescriptorChanged(packet);
+            if (status)
+                *status = ReaderStatus(packet, !invalid).detach();
+
+            if (invalid)
+            {
+                it = packets.erase(packets.begin(), it + 1);
+                cachedSamples -= readCachedSamples;
+            }
+            else 
+                it = packets.erase(it);
```

Comment:
Shouldn't we in all cases clear our history here?

Reply:
I would rather make it configurable